### PR TITLE
Integration test for catalog compilation

### LINKED ISSUE #252: Add integration test for catalog compilation
## Context

We've had a few instances where we've assumed that our changes were correct after all the tests pass. However, we don't have full coverage of everything that needs to go right for `catalog compile` from the tests yet.

It should be doable to implement an integration test by mocking out the calls to the Lieutenant API.

### PR DESCRIPTION
**Pull request by simu**
Resolves #252 

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

**Comment by simu** [comment]
Bit of a mess currently because we need to build the Kapitan helm bindings for the integration test...

**Comment by simu** [comment]
Integration test is currently failing due to not-quite-smart-enough catalog commit message verification.

**Comment by simu** [comment]
> Integration test is currently failing due to not-quite-smart-enough catalog commit message verification.

Ensured catalog commit message is stable by version-pinning components in https://github.com/projectsyn/test-tenant/blob/master/c-test.yml

**Review by ccremer — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -82,6 +82,38 @@ jobs:
...
+    - name: Update PATH
+      run: echo "/opt/bin" >> $GITHUB_PATH
+    - name: Pull in SSH deploy key for integration tests
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+          mkdir -p ~/.ssh
+          ssh-keyscan github.com >> ~/.ssh/known_hosts
+          ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+          ssh-add - <<< "${{ secrets.CATALOG_DEPLOY_KEY }}"
```
[the remark this answers is not in the document]
Comment (simu) [reply .github/workflows/test.yml]: Didn't get it to work using a key which is configured as Github deploy key on the catalog repo. I'll revert the action to use my working step for now.

**Review by corvus-ch — APPROVED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -82,6 +82,38 @@ jobs:
...
+          https://github.com/jsonnet-bundler/jsonnet-bundler/releases/download/v0.4.0/jb-linux-amd64 \
+          && chmod +x /opt/bin/jb
+    - name: Update PATH
+      run: echo "/opt/bin" >> $GITHUB_PATH
+    - name: Pull in SSH deploy key for integration tests
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+          mkdir -p ~/.ssh
+          echo "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==" >> ~/.ssh/known_hosts
```
[the remark this answers is not in the document]
Comment (ccremer) [reply .github/workflows/test.yml:110]: No, please don't. This could open MITM attacks...

[the remark this answers is not in the document]
Comment (simu) [reply .github/workflows/test.yml:110]: I had `ssh-keyscan` originally, but when I was trying to get the `webfactory/ssh-agent` action to work yesterday (unsuccessfully), I read in their README that

> [...] using ssh-keyscan without proper key verification is susceptible to man-in-the-middle attacks. You might prefer putting your known SSH host key in your own Action files to add it to the known_hosts file. The SSH host key is not secret and can safely be committed into the repo.

Naturally, with this approach, the integration test will fail if Github changes their SSH host keys at some point in the future.

[the remark this answers is not in the document]
Comment (ccremer) [reply .github/workflows/test.yml:110]: We've used a fixed GH host key in our internal git.vshn.net repos and they never changed thus far :)
(Yeah, MITM is not huge risk here, but I thought I'd still discourage using ssh-keyscan here)

**Pull request merged**